src/bin/nydusd/fusedev.rs

```diff
@@ -230,8 +237,32 @@ impl NydusDaemon for FusedevDaemon {
     }
 
     fn wait(&self) -> DaemonResult<()> {
+        self.wait_state_machine()?;
+        self.wait_service()
+    }
+
+    fn wait_state_machine(&self) -> DaemonResult<()> {
+        let mut guard = self.state_machine_thread.lock().unwrap();
+        if guard.is_some() {
+            guard
+                .take()
+                .unwrap()
+                .join()
+                .map_err(|e| {
+                    DaemonError::WaitDaemon(
+                        *e.downcast::<Error>()
+                            .unwrap_or_else(|e| Box::new(eother!(e))),
```

Comment:
Um. Hard error conversion path to `io::Error`, maybe let `WaitDaemon` wrap another type of error message is an easier way. But has nothing to do with this PR.